internal/signature/signer.go

```diff
@@ -62,6 +63,14 @@ func Sign(ctx context.Context, req *proto.GenerateSignatureRequest) (*proto.Gene
 		}
 	}
 
+	// Notary to OpenBao/Vault hash algorithm naming conversion map
+	vaultHashAlgorithms := map[string]string{
+		"SHA-256": "sha2-256",
+		"SHA-384": "sha2-384",
+		"SHA-512": "sha2-512",
+	}
+	vaultHashAlgorithm := vaultHashAlgorithms[string(req.Hash)]
```

Comment:
Good catch, I missed this as I wasn't familiar with the request format. :-) 

IIUC, if we fail to find a mapping, we'll default to the empty string, but pass it along. From my read of the [API docs](https://openbao.org/api-docs/secret/transit/#sign-data), this will result in an error, rather than choosing a sane default here. I think that's probably the best option, tbh, but I wonder if it makes sense to err more precisely here rather than take the Vault error?